tests/ui/test_pv_encryption_ui.py

```diff
@@ -0,0 +1,302 @@
+            "Verify whether encrypted device is present inside the pod and run IO"
+        )
+        for vol_handle, pod_obj in zip(vol_handles, pod_objs):
+            if pod_obj.exec_sh_cmd_on_pod(
+                command=f"lsblk | grep {vol_handle} | grep crypt"
+            ):
+                logger.info(f"Encrypted device found in {pod_obj.name}")
+            else:
+                logger.error(f"Encrypted device not found in {pod_obj.name}")
+
+            logger.info(f"Running FIO on Pod '{pod_obj.name}'")
+            pod_obj.run_io(
+                storage_type="block",
+                size=(int(pvc_size) - 1),
+                io_direction="write",
+                runtime=60,
+                invalidate=0,
+            )
+            logger.info(f"IO started on the Pod {pod_obj.name}")
+
+        # Wait for IO completion
+        logger.info("Waiting for IO completion on the Pod")
+        for pod_obj in pod_objs:
+            pod_obj.get_fio_results()
+        logger.info("IO completed on all Pods")
+
+        # Delete the pod
+        for pod_obj in pod_objs:
+            logger.info(f"Deleting Pod '{pod_obj.name}'")
+            pod_obj.delete(wait=True)
+            pod_obj.ocp.wait_for_delete(resource_name=pod_obj.name)
+            logger.info(f"Pod '{pod_obj.name}' Deleted Successfully")
+
+        # Deleting the PVC via UI
+        logger.info(f"Delete pvc '{pvc_name}' via UI")
+        pvc_objs = get_all_pvc_objs(namespace=project_name)
+        pvcs = [pvc_obj for pvc_obj in pvc_objs if pvc_obj.name == pvc_name]
+
+        for pvc in pvc_objs:
+            pvc_ui_obj.delete_pvc_ui(pvc.name, project_name)
+            logger.info(f"Waiting for '{pvc.name}' pvc deletion")
+            pvcs[0].ocp.wait_for_delete(pvc.name, timeout=120)
+
+        logger.info("Checking if pvc's are deleted or not")
+        if len(pvcs) > 0:
+            assert f"PVC {pvcs[0].name} does not deleted"
+
+        # Verify whether the key is deleted in Vault. Skip check for kv-v2 due to BZ#1979244 in OCS 4.8
+        logger.info(
+            "Verify whether the key is deleted in Vault. Skip check for kv-v2 due to BZ#1979244 in OCS 4.8"
+        )
+        ocs_version = version.get_semantic_ocs_version_from_config()
+        if (kv_version == "v1") or (
+            kv_version == "v2" and ocs_version >= version.VERSION_4_9
+        ):
+            for vol_handle in vol_handles:
+                if not kms.is_key_present_in_path(
+                    key=vol_handle, path=self.vault.vault_backend_path
+                ):
+                    logger.info(f"Vault: Key deleted for {vol_handle}")
+                else:
+                    raise KMSResourceCleaneupError(
+                        f"Vault: Key deletion failed for {vol_handle}"
+                    )
+        # Deleting Storage Class via UI
+        logger.info("Deleting Storage Class via UI")
+        sc_deletion_check = sc_obj.delete_rbd_storage_class(sc_name=sc_name)
```

Comment:
If test fails before that storageclass won't be deleted. For this case I've wrote the storageclass ui factory.
You are moving a way from a best practice implementation. 
All you got to do is copy the option of the function `create_encrypted_storage_class_ui` in the class to the factory, check in the factory is kms_address exist and send from there the option to build the storageclass with your new method `create_encrypted_storage_class_ui`.
This way you don't have to handle the teardown and it open to more options to work with a new block pool. 
We can do it together if you want.

Reply:
Done